dashboard/app/models/section.rb

```diff
@@ -60,12 +60,12 @@ def workshop_section?
     Pd::Workshop::SECTION_TYPES.include? section_type
   end
 
-  validates_presence_of :user
+  validates_presence_of :user, unless: -> {deleted?}
   def user_must_be_teacher
     return unless user
     errors.add(:user_id, 'must be a teacher') unless user.teacher?
   end
-  validate :user_must_be_teacher
+  validate :user_must_be_teacher, unless: -> {deleted?}
```

Comment:
Actually it looks like we don't need this because the validation method itself [ignores nil users](https://github.com/code-dot-org/code-dot-org/blob/b193d19af5c0266ac00c886e02ce954e954f8772/dashboard/app/models/section.rb#L65).